pkg/diagnostics/cluster/aggregated_logging/routes.go

```diff
@@ -0,0 +1,95 @@
+package aggregated_logging
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+
+	routes "github.com/openshift/origin/pkg/route/api"
+)
+
+const routeUnaccepted = `
+An unaccepted route is most likely due to one of the following reasons:
+
+* No router has been deployed to serve the route.
+* Another route with the same host already exists.
+
+If a router has been deployed, look for duplicate matching routes by
+running the following:
+
+  oc get --all-namespaces routes --template='{{range .items}}{{if eq .spec.host "%[2]s"}}{{println .metadata.name "in" .metadata.namespace}}{{end}}{{end}}'
+
+`
+const routeCertMissingHostName = `
+Try updating the route certificate to include its host as either the CommonName (CN) or one of the alternate names.
+`
+
+//checkRoutes looks through the logging infra routes to see if they have been accepted, and ...
+func checkRoutes(r diagnosticReporter, adapter routesAdapter, project string) {
+	r.Debug("AGL0300", "Checking routes...")
+	routeList, err := adapter.routes(project, kapi.ListOptions{LabelSelector: loggingSelector.AsSelector()})
+	if err != nil {
+		r.Error("AGL0305", err, fmt.Sprintf("There was an error retrieving routes in the project '%s' with selector '%s'", project, loggingSelector.AsSelector()))
+		return
+	}
+	if len(routeList.Items) == 0 {
+		r.Error("AGL0310", nil, fmt.Sprintf("There were no routes found to support logging in project '%s'", project))
+		return
+	}
+	for _, route := range routeList.Items {
+		if !wasAccepted(r, route) {
+			r.Error("AGL0325", nil, fmt.Sprintf("Route '%s' has not been accepted by any routers."+routeUnaccepted, route.ObjectMeta.Name, route.Spec.Host))
+		}
+		if route.Spec.TLS != nil && len(route.Spec.TLS.Certificate) != 0 && len(route.Spec.TLS.Key) != 0 {
+			checkRouteCertificate(r, route)
+		} else {
+			r.Debug("AGL0331", fmt.Sprintf("Skipping key and certificate checks on route '%s'.  Either of them may be missing.", route.ObjectMeta.Name))
+		}
+	}
+}
+
+func checkRouteCertificate(r diagnosticReporter, route routes.Route) {
+	r.Debug("AGL0330", fmt.Sprintf("Checking certificate for route '%s'...", route.ObjectMeta.Name))
+	block, _ := pem.Decode([]byte(route.Spec.TLS.Certificate))
+	//verify hostname
+	if block != nil {
+		cert, err := x509.ParseCertificate(block.Bytes)
+		if err != nil {
+			r.Error("AGL0335", err, fmt.Sprintf("Unable to parse the certificate for route '%s'", route.ObjectMeta.Name))
```

Comment:
print error in message; this one is actually reasonably likely to happen at some point...